### PR TITLE
Add 3.13 compatibility

### PR DESCRIPTION
**Pull request by benjjs**
Package is currently not compatible with Python 3.13 because it uses 2 functions from the `dataclasses` module that were removed from 3.12 to 3.13.

To resolve this problem, I have simply pulled the relevant functions (along with one helper that was also removed) into the package. I've documented which version they were removed in and the original functions in 3.12 that they are copied from.

The only other change is to explicitly pass empty `type_params` to `typing._eval_type()`, as leaving it implicit raises a warning in 3.13. Given that we previously did not supply this argument and the implicit default value was `()`, explicitly passing `()` will not change the behavior.

There is almost certainly some more elegant way to do this that would involve removing the need for these functions in the first place, but my priority here was to add compatibility quickly and without modifying any underlying logic and hence potentially breaking things. If you (the commenter or reviewer) have the more elegant implementation, by all means put it up and let's do that instead!

### CODE REVIEW

File: dataclass_wizard/utils/typing_compat.py [hunk 1]
```diff
@@ -372,8 +364,12 @@ def eval_forward_ref(base_type: FREF,
     # Evaluate the ForwardRef here
     base_globals = sys.modules[cls.__module__].__dict__
 
-    # noinspection PyProtectedMember
-    return typing._eval_type(base_type, base_globals, _TYPING_LOCALS)
+    if sys.version_info >= (3, 13):
+        # noinspection PyProtectedMember
+        return typing._eval_type(base_type, base_globals, _TYPING_LOCALS, type_params=())
```
Comment (rnag) [inline dataclass_wizard/utils/typing_compat.py:369]: @benjjs Minor thing, can you pull this out to the top level? That way we won't need to do a python version check each time `eval_forward_ref` is called. Thanks!

```
if sys.version_info >= (3, 13):
  _eval_type = functools.partial(typing._eval_type, type_params=())
else:
  _eval_type = typing._eval_type
```

File: dataclass_wizard/utils/compatibility.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+"""
+Pulling some functions removed in recent versions of Python into the module for continued compatibility.
+All function names and bodies are left exactly as they were prior to being removed.
+"""
+
+from dataclasses import MISSING
+from types import FunctionType
+
+
+def _set_qualname(cls, value):
```
Comment (rnag) [inline dataclass_wizard/utils/compatibility.py:10]: Good idea, thanks!

File: dataclass_wizard/serial_json.py [hunk 3]
```diff
@@ -10,6 +8,8 @@
 from .dumpers import asdict
 from .loaders import fromdict, fromlist
 from .type_def import Decoder, Encoder, JSONObject, ListOfJSONObject
+# noinspection PyProtectedMember
+from .utils.compatibility import _create_fn, _set_new_attribute
```
Comment (rnag) [inline dataclass_wizard/serial_json.py:12]: I am wondering if we can come up with a better name for the module. Maybe something like `dataclasses_compat`? I think `compatability` is a bit vague overall.

**Review by rnag — CHANGES_REQUESTED** [review]

**Comment by rnag** [comment]
@benjjs  Thanks so much for creating this PR!

I recently upgraded my laptop to Python 3.13, and I also noticed that Dataclass Wizard breaks on import due to the errors mentioned in the PR. I feel that was my bad on importing hidden functions from (a built in) module, which is a big no-no as I always say 😅 .

Thanks again for your work and effort in creating this PR! I have added some comments, please let me know when you're able to look into the notes I added, and I will merge the bugfix for 3.13 once complete. Have a great day!

File: dataclass_wizard/utils/typing_compat.py (hunk 1 above)
Comment (rnag) [reply dataclass_wizard/utils/typing_compat.py:369]: Follow-up can we please add this to `utils/typing_compat.py` and import/use it instead:

```
# Check if currently running Python 3.8
PY313 = _PY_VERSION == (3, 13)
```

Thanks!

**Comment by benjjs** [comment]
Thanks for reviewing @rnag ! Addressed the above comments.

**Comment by rnag** [comment]
Thanks @benjjs ! Everything looks good to go. I'll review/merge later today.

File: dataclass_wizard/utils/typing_compat.py (hunk 1 above)
Comment (rnag) [reply dataclass_wizard/utils/typing_compat.py:369]: > @benjjs Minor thing, can you pull this out to the top level? That way we won't need to do a python version check each time `eval_forward_ref()` is called. Thanks!
> 
> ```
> if sys.version_info >= (3, 13):
>   _eval_type = functools.partial(typing._eval_type, type_params=())
> else:
>   _eval_type = typing._eval_type
> ```

This was not resolved, but its OK, I can do that. Simple fix.

**Comment by rnag** [comment]
Ok, looks good to go. Thanks! One minor change left, and I will work on the release.

I actually dozed off yesterday night, so I was not able to do it then 😓 . Stay tuned for the release!

**Comment by rnag** [comment]
@benjjs Done - Thanks for your effort and help on this (also your patience!) 

[v0.26.0](https://pypi.org/project/dataclass-wizard/0.26.0/) is officially live, and adds support for PY 3.13 🎉

**Pull request merged**